packages/rum-core/src/domain/resource/matchRequestTiming.ts

```diff
@@ -2,13 +2,16 @@ import type { Duration, RelativeTime } from '@datadog/browser-core'
 import { addDuration } from '@datadog/browser-core'
 import type { RumPerformanceResourceTiming } from '../../browser/performanceCollection'
 import type { RequestCompleteEvent } from '../requestCollection'
-import { toValidEntry } from './resourceUtils'
+import { WeakSet } from '../../browser/polyfills'
+import { isValidEntry } from './resourceUtils'
 
 interface Timing {
   startTime: RelativeTime
   duration: Duration
 }
 
+const alreadyMatchedEntries = new WeakSet<PerformanceEntry>()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** Using a WeakSet for entries looks weird to me, as we don't know for how long the ref will exists since garbage collection can happen at any point in time. To have a more predictable behaviour we could keep the last X entries. Wdyt?‏

Reply:
`performance.getEntriesByName()` is already pulled out from [a buffer](https://developer.mozilla.org/en-US/docs/Web/API/Performance/setResourceTimingBufferSize), so they won't be garbage collected until the buffer is emptied.

If the buffer is emptied before `matchRequestTiming()` is called, then `performance.getEntriesByName()` will not return the entries that we wish to filter out in the first place.

If the garbage collection happens between `performance.getEntriesByName()` and `alreadyMatchedEntries.has()` then the entries we're interested in will be held strongly by `sameNameEntries` and not garbage collected.

In conclusion, I think keep the last X entries will not add more predictability while adding the overhead of managing ourselves a separate buffer.

Does this answer your concern?

Reply:
Very interesting! It's very smart to rely on the refs held by the browser :)